Framework: disable 'source-map-loader' as default

retrofox opened the pull request:
Whit this PR `source-map-loader` stuff is determinate by the feature flag `use-source-maps` and its value is `disabled` as default. To enable run the app doing something like ...

``` cli
ENABLE_FEATURES=use-source-maps make run
```

Keep in mind `source-map-loader` is limited to `development` environment as well.

We've been experiencing high use of the CPU after that `source-map-loader` was added to Webpack. https://github.com/Automattic/wp-calypso/pull/5656

![image](https://cloud.githubusercontent.com/assets/77539/16492713/ff2684ec-3ee2-11e6-996e-53deb74f2740.png)

Test live: https://calypso.live/?branch=update/condition-source-map-loader

retrofox commented:
cc @TooTallNate @mtias @aduth @blowery @gwwar

blowery commented:
I like it! Maybe name the feature `use-source-maps` instead of after the particular loader that webpack uses to pull them in.

retrofox commented:
@blowery updated and rebased.

blowery commented on webpack.config.js:
```diff
@@ -132,6 +125,16 @@ if ( CALYPSO_ENV === 'development' ) {
+
+	if ( config.isEnabled( 'use-source-maps' ) ) {
+		webpackConfig.debug = true;
+		webpackConfig.devtool = '#eval-cheap-module-source-map';
```
might as well go with `eval-source-map` if we're going to make it optional. we'll get column level sourcemaps, which should fix some of the issues @aduth saw with hovering over variable names in the debugger.

blowery commented:
I think we might also want to disable the hot loader when using sourcemaps. That really seems to be where the issue springs up.

retrofox commented:
Updated -> aa9c133179a451ebea2fa8ab03fa9fa6a429a2c2

TooTallNate commented:
Probably a good idea since people are experiencing some issues with it. Darn Chrome, this is why we can't have nice things…

gwwar commented:
Thanks @retrofox, this is a good idea to default this disabled if the utilization is that high.
![giphy](https://cloud.githubusercontent.com/assets/1270189/16497764/79f52768-3ead-11e6-8f0b-3200fc76b4ab.gif)

blowery commented:
We probably want to use the 'eval' devtool like we did before in the default case: https://github.com/Automattic/wp-calypso/pull/5656/files

that gives us per-file breakouts in the debugger, which is better than trying to navigate the code in the individual chunks.

blowery commented:
A tangent, but also noticed that the hotloader _greatly_ increases the build time. Interesting.

retrofox commented:
> I did notice that we no longer have per-file breakouts in the debugger, just the built chunks.

Yes, me too. But it also occurred for me before to these changes.

> We probably want to use the 'eval' devtool like we did before in the default case:

Added and rebased in the first commit

blowery commented:
🚢

retrofox commented:
Thanks for the review guys.

aduth commented on webpack.config.js:
```diff
@@ -4,7 +4,8 @@
 var webpack = require( 'webpack' ),
-	path = require( 'path' );
+	path = require( 'path' ),
+	config = require( 'config' );
```
`config` is already declared 4 lines down from here as `require( './server/config' );`. We should eliminate one.
